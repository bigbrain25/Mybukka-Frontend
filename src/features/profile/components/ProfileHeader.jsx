import React from 'react';

import PropTypes from 'prop-types';
import ProfileHeaderTitle from '../common/ProfileHeaderTitle';

import './profileheader.scss';

const ProfileHeader = ({ firstName, lastName }) => (
  <div className="profile-header-section d-none d-lg-block capitalize">
    <ProfileHeaderTitle firstName={firstName} lastName={lastName} />
  </div>
);

export default ProfileHeader;

ProfileHeader.propTypes = {
  firstName: PropTypes.string.isRequired,
  lastName: PropTypes.string.isRequired
};
