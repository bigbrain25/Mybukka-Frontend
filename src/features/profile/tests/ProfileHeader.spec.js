import React from 'react';
import ProfileHeader from '../components/ProfileHeader';

describe('Profileheader component', () => {
  const props = {
    firstName: 'FirtName',
    lastName: 'lastName',
  };

  const wrapper = mount(<ProfileHeader {...props} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
