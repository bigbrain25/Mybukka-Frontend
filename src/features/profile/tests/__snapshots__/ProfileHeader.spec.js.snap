// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profileheader component renders properly 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <ProfileHeader
    firstName="FirtName"
    lastName="lastName"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "firstName": "FirtName",
      "lastName": "lastName",
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="profile-header-section d-none d-lg-block"
      >
        <div
          class="container profile-header"
        >
          <h1
            class="profile-header-title"
          >
            FirtName
             
            lastName
          </h1>
        </div>
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ProfileHeaderTitle
          firstName="FirtName"
          lastName="lastName"
        />,
        "className": "profile-header-section d-none d-lg-block",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "firstName": "FirtName",
            "lastName": "lastName",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <h1
                className="profile-header-title"
              >
                FirtName
                 
                lastName
              </h1>,
              "classNames": "profile-header",
            },
            "ref": null,
            "rendered": Object {
              "instance": <div
                class="container profile-header"
              >
                <h1
                  class="profile-header-title"
                >
                  FirtName
                   
                  lastName
                </h1>
              </div>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h1
                  className="profile-header-title"
                >
                  FirtName
                   
                  lastName
                </h1>,
                "className": "container profile-header",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <h1
                    class="profile-header-title"
                  >
                    FirtName
                     
                    lastName
                  </h1>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "FirtName",
                      " ",
                      "lastName",
                    ],
                    "className": "profile-header-title",
                  },
                  "ref": null,
                  "rendered": Array [
                    "FirtName",
                    " ",
                    "lastName",
                  ],
                  "type": "h1",
                },
              ],
              "type": "div",
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "firstName": "FirtName",
        "lastName": "lastName",
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="profile-header-section d-none d-lg-block"
        >
          <div
            class="container profile-header"
          >
            <h1
              class="profile-header-title"
            >
              FirtName
               
              lastName
            </h1>
          </div>
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ProfileHeaderTitle
            firstName="FirtName"
            lastName="lastName"
          />,
          "className": "profile-header-section d-none d-lg-block",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "firstName": "FirtName",
              "lastName": "lastName",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <h1
                  className="profile-header-title"
                >
                  FirtName
                   
                  lastName
                </h1>,
                "classNames": "profile-header",
              },
              "ref": null,
              "rendered": Object {
                "instance": <div
                  class="container profile-header"
                >
                  <h1
                    class="profile-header-title"
                  >
                    FirtName
                     
                    lastName
                  </h1>
                </div>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h1
                    className="profile-header-title"
                  >
                    FirtName
                     
                    lastName
                  </h1>,
                  "className": "container profile-header",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <h1
                      class="profile-header-title"
                    >
                      FirtName
                       
                      lastName
                    </h1>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "FirtName",
                        " ",
                        "lastName",
                      ],
                      "className": "profile-header-title",
                    },
                    "ref": null,
                    "rendered": Array [
                      "FirtName",
                      " ",
                      "lastName",
                    ],
                    "type": "h1",
                  },
                ],
                "type": "div",
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
